sync_update/your_updatable_workflow_dacx.go

```diff
@@ -0,0 +1,157 @@
+This practice enables you to add fields in future versions while maintaining backward compatibility.
+You can optionally include a `workflow.Context` parameter in the first position of the function.
+The function can return either a serializable value with an error or just an error.
+
+Update handlers, unlike Query handlers, can observe and mutate Workflow state.
+*/
+
+// YourUpdatableWorkflow is a Workflow Definition.
+// This Workflow sets an Update handler and then sleeps for a minute.
+// After setting the Update hanlder it sleeps for 1 minutue.
+// Updates can be sent to the Workflow during this time.
+func YourUpdatableWorkflow(ctx workflow.Context, param WFParam) (WFResult, error) {
+	counter := param.StartCount
+	workflow.SetUpdateHandler(ctx, YourUpdateName, func(arg YourUpdateArg) (YourUpdateResult, error) {
+		counter += arg.Add
+		result := YourUpdateResult{
+			Total: counter,
+		}
+		return result, nil
+	})
+	// Sleep for 60 seconds to have time to send Updates.
+	workflow.Sleep(ctx, 60*time.Second)
+	endTotal := WFResult{
+		EndTotal: counter,
+	}
+	return endTotal, nil
+}
+
+/*
+Validate certain aspects of the data sent to the Workflow using an Update validator function.
+For instance, a counter Workflow might never want to accept a non-positive number.
+Invoke the `SetUpdateHandlerWithOptions` API and define a validator function as one of the options.
+
+When you use a Validator function, the Worker receives the Update first, before any Events are written to the Event History.
+If the Update is rejected, it's not recorded in the Event History.
+If it's accepted, the `WorkflowExecutionUpdateAccepted` Event occurs.
+Afterwards, the Worker executes the accepted Update and, upon completion, a `WorkflowExecutionUpdateCompleted` Event gets written into the Event History.
+*/
+
+// UpdatableWorkflowWithValidator is a Workflow Definition.
+// This Workflow Definition has an Update handler that uses the isPositive() validator function.
+// After setting the Update hanlder it sleeps for 1 minutue.
+// Updates can be sent to the Workflow during this time.
+func UpdatableWorkflowWithValidator(ctx workflow.Context, param WFParam) (WFResult, error) {
+	counter := param.StartCount
+	if err := workflow.SetUpdateHandlerWithOptions(
+		ctx, YourValidatedUpdateName,
+		func(arg YourUpdateArg) (YourUpdateResult, error) {
+			counter += arg.Add
+			result := YourUpdateResult{
+				Total: counter,
+			}
+			return result, nil
+		},
+		// Set the isPositive validator.
+		workflow.UpdateHandlerOptions{Validator: isPositive},
+	); err != nil {
+		return WFResult{}, err
+	}
+	// Sleep for 60 seconds to have time to send Updates.
+	workflow.Sleep(ctx, 60*time.Second)
+	endTotal := WFResult{
+		EndTotal: counter,
+	}
+	return endTotal, nil
+}
+
+// isPositive is a validator function.
+// It returns an error if the int value is below 1.
+func isPositive(ctx workflow.Context, u YourUpdateArg) error {
```

Comment:
Strictly speaking the workflow.Context parameter is optional for both the validation function and the execution function. And if it's present in one but not the other, that's fine. But from an example standpoint it's probably more clear if it's either in both or neither.